### ISSUE TITLE
Same speed of training Tabular NN using 4 CPU and 32 CPU computer 

### ISSUE THREAD

**Issue by panchekha** [awslabs/autogluon#1534]
For a fixed test NN 
both c5.2xlarge (4 CPU) and c5.18xlarge (32 CPU) 
complete fit of 10 epochs in the same (~ 1.5 minute)

How come?

**Comment by willsmithorg** [comment]
Could you provide more details please:
- rough size of dataset, rows and columns, and what types are the features
- parameters you supplied to `autogluon.tabular`
- results from fit() and leaderboard() from both VM sizes

Personally I see most of the models within AutoGluon are using all of the 32 cores on my local machine, so this is not expected behaviour - but I'm mostly using boosted trees not NN.

**Comment by panchekha** [comment]
Important: based on TOP, all CPUs being used, but overall, this fixed size fitting takes almost exactly the same time on 4CPU 32CPU.

Type: Tabular
Database: 200,000 rows
Each row is 20 features

hyperparameters = {  'NN': nn_options }
hyperparameter_tune_kwargs = {  
                'num_trials': 1,
                'scheduler' : 'local',
                'searcher'  : 'auto',
                'hyperparameter_tune':True,
            }
ag_args_fit = {
                'num_gpus': None,
                'num_cpus': None,
            }
predictor = TabularPredictor( label="Target", path="model").fit(train_data,
            	auto_stack    = False, 
            	refit_full    = False,
                hyperparameters            = hyperparameters,
                hyperparameter_tune_kwargs = hyperparameter_tune_kwargs,
                ag_args_fit                = ag_args_fit,
                holdout_frac  = 0.2,
                verbosity     = 4,
                time_limit    = 24*60*60,
            	)

[leaderboard]
,model,score_val,pred_time_val,fit_time,pred_time_val_marginal,fit_time_marginal,stack_level,can_infer,fit_order
0,NeuralNetMXNet,-2.59021277796911,1.4330658912658691,81.66375303268433,1.4330658912658691,81.66375303268433,1,True,1
1,WeightedEnsemble_L2,-2.59021277796911,1.4352188110351562,81.6663990020752,0.0021529197692871094,0.0026459693908691406,2,True,2

**Comment by Innixma** [comment]
Due to the small size of tabular neural networks, they do not benefit much from parallel cores during fitting. This is an ongoing area of optimization we are interested in, so if anyone can identify and provide a working example of a more performant tabular neural network, we would be very interested in it. In v0.4, Tab NN models will be ~10x faster to train in bagged mode as we parallelize the training of the fold models, meaning we train 10 NN models at the same time. This will not speed up the default setting however, as the wall-clock time to train 1 NN model will stay the same.

For now I am closing as this behavior is expected. Please re-open if you find a separate Tabular NN model that is able to effectively utilize large amounts of cores.
